### PR TITLE
Groups: support additional settings for unified groups

### PR DESCRIPTION
**Pull request by manicminer**
* `data.azuread_group` - support for the `allow_external_senders`, `auto_subscribe_new_members`, `hide_from_address_lists` and `hide_from_outlook_clients` attributes
* `azuread_group` - support for the `allow_external_senders`, `auto_subscribe_new_members`, `hide_from_address_lists` and `hide_from_outlook_clients` properties

Note: depends on https://github.com/manicminer/hamilton/pull/143, requires rebase

Closes: #719

### CODE REVIEW

**Comment by manicminer** [comment]
**Test Results**

Some CI issues at the moment, also the newly introduced tests cannot yet run in CI

Screenshot 2022-01-28 at 01 52 52

Screenshot 2022-01-28 at 01 07 16

File: docs/data-sources/group.md [hunk 1]
```diff
@@ -38,11 +38,15 @@ The following arguments are supported:
 
 The following attributes are exported:
 
+* `allow_external_users` - Indicates whether people external to the organization can send messages to the group. Only set for Unified groups.
```
Comment (katbyte) [inline docs/data-sources/group.md:41]: should this be
```suggestion
* `external_users_allowed` - Indicates whether people external to the organization can send messages to the group. Only set for Unified groups.
```
to match the tense of other properties

**Review by katbyte — APPROVED** [review]
aside from 1 comment LGTM ⛵

File: docs/data-sources/group.md (hunk 1 above)
Comment (manicminer) [reply docs/data-sources/group.md:41]: Good catch, will update 👍

**Pull request merged**